### DISCUSSION TITLE
Improve dataset icons in explorer view

### DISCUSSION THREAD

**Discussion by eugeniamz** [apache/superset#22484] [Ideas]
In the dataset window we have different icons for virtual dataset and physical tables 
<img width="795" alt="Screen Shot 2020-12-23 at 7 21 56 AM" src="https://user-images.githubusercontent.com/58375897/102995980-2fd10780-44f0-11eb-9aaa-890cb3662d0a.png">

When we create a dataset we have different icon to see if a table is a physical table or a database view 
<img width="624" alt="Screen Shot 2020-12-23 at 7 22 17 AM" src="https://user-images.githubusercontent.com/58375897/102996032-4d9e6c80-44f0-11eb-9e8a-d5de0c6a3a3b.png">

But the explorer window we always shows the same icon 
<img width="702" alt="Screen Shot 2020-12-23 at 7 20 39 AM" src="https://user-images.githubusercontent.com/58375897/102996053-5bec8880-44f0-11eb-8cc6-986dd7134881.png">

Is it possible 
1- To add the view icon in the dataset to distinguish physical tables and physical views 
2- Add the 3 different icons on the explorer view.  If one is not possible at least add a different icon for virtual and physical. 

As a data engineer, knowing the source of the data with a simple visual instead to do a search is very useful.

**Comment by betodealmeida** [comment]
This is something we should improve because the messaging is not clear (cc: @srinify).

These are different things: inside the database we have two types of **tables**, physical tables and (native) views. In SQL Lab these are represented by the table icon and the eye icon, respectively.

Once you're outside SQL Lab, Superset doesn't really care if the database entity is a physical table or a native view — it treats them the same, because the DB abstract these two entities through a single API: SQL.

Superset has a concept that lives atop the physical tables and native views: the dataset. The dataset can be physical or virtual. A **physical dataset** is one that points to a "table name" on the DB, that could be physical or a view. A **virtual dataset** is one that has SQL associated with it, pretty much like a native DB view. The main difference is that the SQL defining a virtual dataset lives in the Superset main database.

So you could have a **physical dataset** in Superset that points to a **native DB view**. Or a **virtual dataset** that is defined by SQL stored in Superset, pointing to a **physical dataset** that points to a DB native view, defined by SQL stored in the DB.

In summary, at the DB level we have 2 things: physical tables and views.
One level above, we have Superset datasets, that can be: physical datasets or virtual datasets.

This is super confusing, and we should figure out a better way of communicating this. Maybe we can use native views whenever possible, so that when a user clicks "visualize" in SQL Lab we run CVAS if the DB supports it? And then we provide ways of fetching the view definition stored in the DB so that the user can edit it.

**Comment by srinify** [comment]
Somewhat related, I find myself having to hover over the icon frequently because the icons themselves don't clearly scream "physical" or "virtual" (in the way that the eye / eyeball icon actually does imply virtual).

**Comment by rusackas** [comment]
This thread hasn't been touched in a while, but it's still a pertinent discussion! Pinging @jess-dillard here to see if there's a good way to re-route this. That said, I'm also going to close the GH Issue (it's not a bug) and move it to an Ideas thread on Discussions, so we can take up the topic there! Thanks!
